libsmtutil/CHCSmtLib2Interface.cpp

```diff
@@ -115,24 +133,18 @@ void CHCSmtLib2Interface::declareVariable(std::string const& _name, SortPointer
 	else if (!m_variables.count(_name))
 	{
 		m_variables.insert(_name);
-		write("(declare-var |" + _name + "| " + toSmtLibSort(*_sort) + ')');
+		write("(declare-var |" + _name + "| " + toSmtLibSort(_sort) + ')');
 	}
 }
 
-std::string CHCSmtLib2Interface::toSmtLibSort(Sort const& _sort)
+std::string CHCSmtLib2Interface::toSmtLibSort(SortPointer _sort)
 {
-	if (!m_sortNames.count(&_sort))
-		m_sortNames[&_sort] = m_smtlib2->toSmtLibSort(_sort);
-	return m_sortNames.at(&_sort);
+	return m_smtlib2->toSmtLibSort(_sort);
 }
 
 std::string CHCSmtLib2Interface::toSmtLibSort(std::vector<SortPointer> const& _sorts)
 {
-	std::string ssort("(");
-	for (auto const& sort: _sorts)
-		ssort += toSmtLibSort(*sort) + " ";
-	ssort += ")";
-	return ssort;
+	return m_smtlib2->toSmtLibSort(_sorts);
```

Comment:
We could just use `m_smtlib2->toSmtLibSort(_sorts)` instead of having function `CHCSmtLib2Interface::toSmtLibSort`.